### PR TITLE
authz: fix authz policy to use path matcher

### PR DESCRIPTION
**Pull request by yangminzhu**
Cherry-pick https://github.com/istio/istio/pull/21132 to master

### CODE REVIEW

**Comment by yangminzhu** [comment]
It seems the recent proxy sha update in master still doesn't include the needed upstream Envoy changes, need to update the Envoy first.

File: pilot/pkg/security/authz/model/matcher/header_test.go [hunk 1]
```diff
@@ -59,3 +60,78 @@ func TestHeaderMatcher(t *testing.T) {
...
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			actual := PathMatcher(tc.V)
+			if !reflect.DeepEqual(*tc.Expect, *actual) {
+				t.Errorf("%s: expecting %v, but got %v", tc.Name, *tc.Expect, *actual)
```
Comment (diemtvu) [inline pilot/pkg/security/authz/model/matcher/header_test.go]: nit. Name is already shown in the test case, you can skip it in the error message.

File: pilot/pkg/security/authz/model/model.go [hunk 2]
```diff
@@ -68,6 +68,7 @@ const (
 	// Envoy config attributes for ServiceRole rules.
 	methodHeader = ":method"
 	pathHeader   = ":path"
```
Comment (diemtvu) [inline pilot/pkg/security/authz/model/model.go]: Does this still needed, if so, in what cases? Maybe add some comment to be clear.

Comment (yangminzhu) [reply pilot/pkg/security/authz/model/model.go]: cleaned other code as well.

**Comment by yangminzhu** [comment]
Blocked by https://github.com/istio/proxy/pull/2725

**Pull request merged**